### DISCUSSION TITLE
How do you find the actual song time of a note?

### DISCUSSION THREAD

**Discussion by dreammixgames** [melanchall/drywetmidi#155] [Q&A] [answered]
I was loading notes and dividing them by the current beats per minute to get the note position before. I found out that that wouldn't work how I wanted it to. What I want to know how to do now, is how to find out the actual time that the note is played, and not Note.Time. Is that possible?

**Comment by melanchall** [comment, marked as answer]
You can get time and length in many different formats. Please see [overview article](https://melanchall.github.io/drywetmidi/articles/high-level-managing/Time-and-length-overview.html) of the documentation. For example, to get a note's time in terms of microseconds, use this code:

```csharp
var metricTimeSpan = note.TimeAs<MetricTimeSpan>(tempoMap);
```

Just use [documentation](https://melanchall.github.io/drywetmidi/api/Melanchall.DryWetMidi.Interaction.TimedObjectUtilities.html#Melanchall_DryWetMidi_Interaction_TimedObjectUtilities_TimeAs__1_Melanchall_DryWetMidi_Interaction_ITimedObject_Melanchall_DryWetMidi_Interaction_TempoMap_) to learn more about API.

**Reply by dreammixgames** [reply]
Thanks! I figured that out!
